lib/game.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require_relative 'user'
+
+class Game
+  attr_reader :total_turns
+
+  def initialize(userx = User.new('X'), usero = User.new('O'))
+    @user = {
+      x: userx,
+      o: usero
+    }
+    reset_game
+  end
+
+  private
+
+  def winner(last_p)
+    current_value = @turn_x ? 'O' : 'X'
+    winner_cases = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+    winner_cases.each do |x|
+      next unless x.include? last_p
+      if current_value == @board[x[0]] && current_value == @board[x[1]] && current_value == @board[x[2]]
+        return current_value
+      end
+    end
+    false
+  end
+
+  public
+
+  def move_game(position) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return "Position occupied with #{@board[position - 1]}" unless @board[position - 1].nil?
+    return 'Position out of boundaries. Please numbers from 1 to 9' unless position >= 1 && position <= 9
+
+    @board[position - 1] = @turn_x ? 'X' : 'O'
+    @turn_x = !@turn_x
+    @total_turns -= 1
+    winner = winner(position - 1)
+
+    return "It's a draw" if !winner && total_turns < 1
+
+    return winner unless winner
+
+    @total_turns = 0 if winner
+    return "The champion is #{@user[winner.downcase.to_sym].name} '#{winner}'!!" if winner
+  end
```

Comment:
**`OPTIONAL`**
Don't return a string value. Instead, return a value and use that value to determine the message to print to the user

For example, you could write it like this
```ruby
def move_game(position)
    return :INVALID_POSITION_OCCUPIED unless @board[position - 1].nil?
    return :INVALID_POSITION_OUT_BOUNDARIES unless position >= 1 && position <= 9

    @board[position - 1] = @turn_x ? 'X' : 'O'
    @turn_x = !@turn_x
    @total_turns -= 1
    winner = winner(position - 1)

    return :DRAW if !winner && total_turns < 1

    return winner unless winner

    @total_turns = 0 if winner
    return :CHAMPION if winner
  end
```

Try to do something like this, please.